Sources/TuistGenerator/Utils/SwiftPackageManagerInteractor.swift

```diff
@@ -0,0 +1,64 @@
+import Basic
+import Foundation
+import TuistCore
+import TuistSupport
+
+public protocol SwiftPackageManagerInteracting {
+    func install(graph: Graphing, workspaceName: String) throws
+}
+
+public class SwiftPackageManagerInteractor: SwiftPackageManagerInteracting {
+    private let fileHandler: FileHandling
+    public init(fileHandler: FileHandling = FileHandler.shared) {
+        self.fileHandler = fileHandler
+    }
+
+    public func install(graph: Graphing, workspaceName: String) throws {
+        try generatePackageDependencyManager(at: graph.entryPath,
+                                             workspaceName: workspaceName,
+                                             graph: graph)
+    }
+
+    private func generatePackageDependencyManager(
+        at path: AbsolutePath,
+        workspaceName: String,
+        graph: Graphing
+    ) throws {
+        let packages = graph.packages
+
+        if packages.isEmpty {
+            return
+        }
+
+        let hasRemotePackage = packages.first(where: { node in
```

Comment:
Should we extract this into an extension for Array where the Element are packages? *Adding some pseudocode below:*

```swift
extension Array where Element: Package {

}
```